cmd/internal/compress/compress.go

```diff
@@ -0,0 +1,90 @@
+package compress
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"github.com/metal-stack/backup-restore-sidecar/cmd/internal/constants"
+	"github.com/mholt/archiver/v3"
+)
+
+type (
+	// Compressor compress/decompress backup data before/after sending/receiving from storage
+	Compressor struct {
+		method Method
+	}
+
+	// Method defines possible compression methods
+	Method uint
+)
+
+const (
+	// TAR all files without compression, is suitable if content is already compressed
+	TAR Method = iota
+	// TARGZ compression
+	TARGZ
+	// TARLZ4 is much faster than GZIP with slightly bigger files
+	TARLZ4
+)
+
+// New Returns a new Compressor with this method
+func New(method Method) *Compressor {
```

Comment:
How about passing the method string from viper here and doing the type switch from `MethodFrom` in the constructor, then return an error if the method is unsupported? This way, you can store the compression function, extension, etc. in the Compressor's private fields and you do not need to type `switch` more than once? Fewer "only-theoretical" errors could occur (like "unknown compression method:%d") and things go wrong immediately in the constructor (which I think is a little better).